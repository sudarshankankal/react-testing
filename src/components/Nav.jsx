import React from "react";
import { Container, Navbar } from "react-bootstrap";

function Nav() {
  return (
    <Navbar bg="light">
      <Container>
        <Navbar.Brand>React Unit Testing (Jest/Enzyme)</Navbar.Brand>
      </Container>
    </Navbar>
  );
}

export default Nav;
