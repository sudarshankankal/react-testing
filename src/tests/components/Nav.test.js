import React from "react";
import { mount, shallow } from "enzyme";
import Nav from "../../components/Nav";
import { Navbar } from "react-bootstrap";


describe("Given Nav", () => {

  test("Match Nav Snapshot", () => {
      const wrapper = shallow(<Nav/>);
      expect(wrapper).toMatchSnapshot();
  })  

  test("Title should render properly", () => {
        const wrapper = shallow(<Nav/>);
        expect(wrapper.contains(<Navbar.Brand>React Unit Testing (Jest/Enzyme)</Navbar.Brand>)).toBe(true);
  });
});
