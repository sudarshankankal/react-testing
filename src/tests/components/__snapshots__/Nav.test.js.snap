// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given Nav Match Nav Snapshot 1`] = `
<Navbar
  bg="light"
  collapseOnSelect={false}
  expand={true}
  variant="light"
>
  <Container
    fluid={false}
  >
    <NavbarBrand>
      React Unit Testing (Jest/Enzyme)
    </NavbarBrand>
  </Container>
</Navbar>
`;
